backend/authentication/providers/basic/basic_test.go

```diff
@@ -0,0 +1,205 @@
+package basic
+
+import (
+	"context"
+	"testing"
+
+	corev2 "github.com/sensu/sensu-go/api/core/v2"
+	"github.com/sensu/sensu-go/backend/authentication/bcrypt"
+	"github.com/sensu/sensu-go/backend/store"
+	storev2 "github.com/sensu/sensu-go/backend/store/v2"
+	"github.com/sensu/sensu-go/testing/mockstore"
+	"github.com/stretchr/testify/mock"
+)
+
+func TestProviderAuthenticate(t *testing.T) {
+	tests := []struct {
+		Name     string
+		Store    storev2.Interface
+		Username string
+		Password string
+		Hash     string
+		Error    bool
+	}{
+		{
+			Name:     "username not supplied",
+			Username: "",
+			Password: "hello",
+			Error:    true,
+		},
+		{
+			Name:     "password not supplied",
+			Username: "Hello",
+			Password: "",
+			Error:    true,
+		},
+		{
+			Name: "user not found",
+			Store: func() storev2.Interface {
+				s := new(mockstore.V2MockStore)
+				s.On("Get", mock.Anything, mock.Anything).Return(nil, &store.ErrNotFound{})
+				return s
+			}(),
+			Username: "eric",
+			Password: "password",
+			Error:    true,
+		},
+		{
+			Name: "user disabled",
+			Store: func() storev2.Interface {
+				s := new(mockstore.V2MockStore)
+				user := &corev2.User{
+					Username: "eric",
+					Disabled: true,
+				}
+				s.On("Get", mock.Anything, mock.Anything).Return(mockstore.Wrapper[*corev2.User]{Value: user}, nil)
+				return s
+			}(),
+			Username: "eric",
+			Password: "password",
+			Error:    true,
+		},
+		{
+			Name: "password hash does not match",
+			Store: func() storev2.Interface {
+				s := new(mockstore.V2MockStore)
+				pw, err := bcrypt.HashPassword("password")
+				if err != nil {
+					panic(err)
+				}
+				user := &corev2.User{
+					Username:     "eric",
+					PasswordHash: pw,
+				}
+				s.On("Get", mock.Anything, mock.Anything).Return(mockstore.Wrapper[*corev2.User]{Value: user}, nil)
+				return s
+			}(),
+			Username: "eric",
+			Password: "asldjflkdsajfl",
+			Error:    true,
+		},
+		{
+			Name: "password hash does match",
+			Store: func() storev2.Interface {
+				s := new(mockstore.V2MockStore)
+				pw, err := bcrypt.HashPassword("password")
+				if err != nil {
+					panic(err)
+				}
+				user := &corev2.User{
+					Username:     "eric",
+					PasswordHash: pw,
+				}
+				s.On("Get", mock.Anything, mock.Anything).Return(mockstore.Wrapper[*corev2.User]{Value: user}, nil)
+				return s
+			}(),
+			Username: "eric",
+			Password: "password",
+			Error:    false,
+		},
+	}
```

Comment:
Should you add a test to cover that branch for the legacy case where, if `User.PasswordHash == ""` we use `User.Password` instead?

Reply:
That's a good question. I thought about the legacy case while I was working on this stuff. I concluded that we should probably drop core/v2.User and add core/v3.User instead to drop support for the legacy case. But I couldn't do that in this PR... I got stuck elsewhere in the system while testing.

What are your thoughts on preserving the legacy case? IMO it is harmful.

Reply:
I think moving from `core/v2.User` to `core/v3.User` is indeed a separate task; let's keep it separate in the interest of not having the scope of this refactor explode.

My only thought right now about that legacy `Password` field in that removing it and support for it sounds good from a technical standpoint, but I would first check what that means for users and operators, especially those of have a db that has been migrated over and over.

It could be as simple as adding a migration, but I'm not 100% sure.

For now, I think the ideal course of action is to add a quick test case for it, for the sake of coverage, and it will fail to build and get removed when we do that v2 -> v3 work?

Reply:
There won't be any migration from 6 to 7, it'll be greenfield databases only.